### DISCUSSION TITLE
Compiling for Windows 11 23H2 with Statically Linked Libraries in Microsoft Visual Studio 2022

### DISCUSSION THREAD

**Discussion by mydarkthawts** [GiacomoLaw/Keylogger#164] [General]
### What and Why?
I figured with the numerous user error issue requests being opened over compilation and dll issues, I'd make a quick text and video guide on how to compile this on Windows 11 23H2 with Statically Linked Libraries in Microsoft Visual Studio 2022. 

### Use Release instead of Debug when compiling
Some people will run into .dll errors that they aren't paying attention to closely enough. It has the letter D in the .dll filename error popup. 
This indicates the executable was compiled with a Debug configuration and uses different runtime libraries for debugging purposes. These debug dynamic linked libraries are not installed with the runtime libraries that people install on target systems. Instead, they exist in your development environment only.

**The fix (alternative seen in my video below):**
1. Click on the `Build` menu near the top of Visual Studio 2022 window.
2. Select `Configuration Manager` near the bottom of the menu.
3. In the `Active solution configuration:` drop down menu, select `Release`.
4. Click `Close`.
5. Build your project.

Your compiled executable should now run with the runtime libraries installed on the target system without .dll errors.

### Statically Linked Libraries instead of using Dynamically Linked Libraries?
Compiling with statically linked libraries allows the executable to run without the need of installing the runtime dependency libraries on the target system(removes the risk of those pesky .dll errors). 
The final compiled executable is generally larger in size because it carries with it the dependent libraries it needs to run to avoid those dll errors.

**This is seen in my video below:**
1. Project Properties > C/C++ > Code Generation > Runtime Library
2. Select `Multi-threaded (/MT)` in the drop down.
3. Apply your changes.
4. Build your project.

### Request Admin Privs on Program execution?
You can edit the project settings and configure it to request administrative privileges at execution time if you are having trouble logging keys from certain applications:

**Not seen in my video below:**
1. Project Properties > Linker > Manifest File > UAC Execution Level
2. Select `requireAdministrator (/level='requireAdministrator')` in the drop down. 
4. Apply your changes.
5. Build your project.

### Video

https://github.com/user-attachments/assets/3fff8886-28b9-4343-ab23-56effa85d9aa

**Comment by USA-good** [comment]
Thanks
